File: src/main/java/ar/edu/um/tesoreria/rest/repository/IValorMovimientoRepository.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.repository;

import java.time.OffsetDateTime;
import java.util.List;
import java.util.Optional;

import ar.edu.um.tesoreria.rest.kotlin.model.ValorMovimiento;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.stereotype.Repository;

/**
 * @author daniel
 *
 */
@Repository
public interface IValorMovimientoRepository extends JpaRepository<ValorMovimiento, Long> {

	public List<ValorMovimiento> findAllByValorMovimientoIdIn(List<Long> valorMovimientoIds);

	public Optional<ValorMovimiento> findByValorIdAndNumero(Integer valorId, Long numero);

	public Optional<ValorMovimiento> findByValorMovimientoId(Long valorMovimientoId);

	public Optional<ValorMovimiento> findFirstByFechaContableAndOrdenContable(OffsetDateTime fechaContable,
			Integer ordenContable);

	@Modifying
	public void deleteByValorMovimientoId(Long valorMovimientoId);
}

File: src/main/java/um/tesoreria/rest/kotlin/repository/IValorMovimientoRepository.kt
package um.tesoreria.rest.kotlin.repository

import org.springframework.data.domain.Sort
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import um.tesoreria.rest.kotlin.model.ValorMovimiento
import java.time.OffsetDateTime
import java.util.Optional

interface IValorMovimientoRepository : JpaRepository<ValorMovimiento, Long> {

    fun findAllByValorMovimientoIdIn(valorMovimientoIds: List<Long?>?): List<ValorMovimiento?>?

    fun findAllByValorMovimientoIdInAndEstadoNotAndEstadoNot(valorMovimientoIds: List<Long?>?, Anulado: String, Reemplazado: String, sort: Sort): List<ValorMovimiento?>?

    fun findFirstByValorIdAndNumero(valorId: Int?, numero: Long?): Optional<ValorMovimiento?>?

    fun findFirstByValorIdAndNumeroAndBancariaIdOrigen(valorId: Int?, numero: Long?, bancariaId: Long?): Optional<ValorMovimiento?>?

    fun findByValorMovimientoId(valorMovimientoId: Long?): Optional<ValorMovimiento?>?

    fun findFirstByFechaContableAndOrdenContable(fechaContable: OffsetDateTime?, ordenContable: Int?): Optional<ValorMovimiento?>?

    @Modifying
    fun deleteByValorMovimientoId(valorMovimientoId: Long?)

}